spec/operation_needs_spec.cr

```diff
@@ -20,27 +20,7 @@ private class NeedsSaveOperation < Needs::SaveOperation
   needs optional : String = "bar"
 end
 
-private class NeedsWithOnOptionSaveOperation < Needs::SaveOperation
-  needs created_by : String, on: :create
-  needs updated_by : String, on: :update
-  needs saved_by : String, on: :save
```

Comment:
Needs with on is rarely used and it is confusing since if you use `on` it means the type must be nilable. This is not ideal and caused more complexity in the macros. So we decided to remove it